### ISSUE TITLE
Make cuts GUI work with properties

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#10]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/10) by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Galen turned MicrocalDataset.p_peak_time from a vector to a property.  Smart, but it crashes the cuts GUI, because the latter expects it to appear in MicrocalDataset.__dict__.keys(), and it doesn't.

I think I know how to fix.

```
#!python

In [675]: cuts=mass.gui.create_cuts(data)
800 1024 100.0 xxxx
0 -3725.17 12455.1
1 1.18003 3813.11
2 509.792 15909.3
3 7 13722
4 -7.93447 1445.92
5 -9.9e-06 2.2669
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)

/home/fowlerj/riken/<ipython console> in <module>()

/usr/local/lib/python2.6/dist-packages/mass/gui/make_cuts_dialog.pyc in create_cuts(datagroup, existing_cuts)
     47     Returns: a mass.TESGroup object, or (if user cancels or selects no files) None.
     48     """
---> 49     dialog = _CutsCreator(datagroup, existing_cuts=existing_cuts)
     50     retval = dialog.exec_()
     51     if retval == _CutsCreator.Rejected:

/usr/local/lib/python2.6/dist-packages/mass/gui/make_cuts_dialog.pyc in __init__(self, datagroup, existing_cuts)
    185                                         "p_peak_value","p_max_posttrig_deriv",
    186                                         "p_rise_time","p_peak_time")):
--> 187             self.cuts[i].compute_actual_range((ds.__dict__[vector_name] for ds in self.data.datasets))
    188             print i, self.cuts[i].actual_min, self.cuts[i].actual_max
    189 

/usr/local/lib/python2.6/dist-packages/mass/gui/make_cuts_dialog.pyc in compute_actual_range(self, data_vectors)
    110     def compute_actual_range(self, data_vectors):
    111         a,b = 9e99, -9e99
--> 112         for v in data_vectors:
    113             a = min(a, v.min())
    114             b = max(b, v.max())

/usr/local/lib/python2.6/dist-packages/mass/gui/make_cuts_dialog.pyc in <genexpr>((ds,))
    185                                         "p_peak_value","p_max_posttrig_deriv",
    186                                         "p_rise_time","p_peak_time")):
--> 187             self.cuts[i].compute_actual_range((ds.__dict__[vector_name] for ds in self.data.datasets))
    188             print i, self.cuts[i].actual_min, self.cuts[i].actual_max
    189 

KeyError: 'p_peak_time'
> /usr/local/lib/python2.6/dist-packages/mass/gui/make_cuts_dialog.py(187)<genexpr>()
    186                                         "p_rise_time","p_peak_time")):
--> 187             self.cuts[i].compute_actual_range((ds.__dict__[vector_name] for ds in self.data.datasets))
    188             print i, self.cuts[i].actual_min, self.cuts[i].actual_max

ipdb> 

```

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Change handling of computing actual range of pulse parameter vectors.  
Instead of finding the vectors in ds.__dict__, which doesn't work when
the vector is a python property, we exec a statement that works either way.
I think this fixes Issue 10: Make cuts GUI work with properties.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Changed handling of pulse summary data from assuming the vectors are all stored in `MicrocalDataset.__dict__` (which properties aren't) to running a full 
`exec("...")` statement.  Fixed in commit dcf41cf9dc6f610.
